### ISSUE TITLE
Ability to pass custom context while executing a query

### ISSUE THREAD

**Issue by mehtasankets** [graphql-java-kickstart/graphql-java-servlet#80]
While implementing dataloaders with graphql-java, there is a usecase to pass loaders as a context to executionInput:

```
DataLoaderRegistry loaders = ...; //Initializing dataloaders
GraphQL runtime = GraphQL.newGraphQL(schemaFromSPQR)
          .instrumentation(new DataLoaderDispatcherInstrumentation(loaders))
          .build();
graphQL.execute(ExecutionInput.newExecutionInput()
         .query(operation)
         .context(loaders) // <-- Need to pass loaders as a context.
         .build());
```

This is required so that loaders can be accessed from resolver like:
```
@GraphQLQuery
public CompletableFuture<Item> item(@GraphQLArgument(name = "id") Long id, 
@GraphQLRootContext DataLoaderRegistry loaders){ 
      return loaders.getDataLoader("items").load(id); //load using the injected DataLoader
}
```

But, I didn't find a way to pass custom context to graphql-java-servlet while resolving a query.

Can we extend GraphQLContext to accept custom context object?

**Comment by ashu-walmart** [comment]
I encountered a similar issue using graphql-servlet and spring-boot-starter. The issue is being able to correctly scope the DataLoaderRegistry to get request level caching with data loaders. Maybe adding Instrumentation or Dataloader as an Optional to GraphQLContext can help. Perhaps something along the lines of this patch attached as text.

[graphql-servlet-1.patch.txt](https://github.com/graphql-java/graphql-java-servlet/files/2094404/graphql-servlet-1.patch.txt)

**Comment by ashu-walmart** [comment]
I added a PR for this. Instead of the entire instrumentation. Am looking to add only the DataLoaderRegistry that can be pulled out of the context. Here is the PR: https://github.com/graphql-java/graphql-java-servlet/pull/83/files

**Comment by oliemansm** [comment]
Closing this issue since this seems to have been addressed when we merged #83. Please reopen @mehtasankets if you feel there's still something that needs to be updated here.
